### PR TITLE
Adding iperf test

### PR DESCRIPTION
**Pull request by narasimhan-v**
Added iperf network performance test.

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
```
2018-06-20 03:46:40,590 parameters       L0146 DEBUG| PARAMS (key=EXPECTED_THROUGHPUT, path=*, default=85) => 95
2018-06-20 03:46:40,590 process          L0554 INFO | Running 'iperf3 -c 9.47.78.174'
2018-06-20 03:46:50,773 process          L0385 DEBUG| [stdout] Connecting to host 9.47.78.174, port 5201
2018-06-20 03:46:50,773 process          L0642 INFO | Command 'iperf3 -c 9.47.78.174' finished with 0 after 10.1802361012s
2018-06-20 03:46:50,773 process          L0385 DEBUG| [stdout] [ 11] local 9.47.66.117 port 39794 connected to 9.47.78.174 port 5201
2018-06-20 03:46:50,774 process          L0385 DEBUG| [stdout] [ ID] Interval           Transfer     Bitrate         Retr  Cwnd
2018-06-20 03:46:50,774 process          L0385 DEBUG| [stdout] [ 11]   0.00-1.00   sec   113 MBytes   944 Mbits/sec    0    363 KBytes       
2018-06-20 03:46:50,774 process          L0385 DEBUG| [stdout] [ 11]   1.00-2.00   sec   112 MBytes   938 Mbits/sec    0    363 KBytes       
2018-06-20 03:46:50,774 process          L0385 DEBUG| [stdout] [ 11]   2.00-3.00   sec   113 MBytes   945 Mbits/sec    0    363 KBytes       
2018-06-20 03:46:50,775 process          L0385 DEBUG| [stdout] [ 11]   3.00-4.00   sec   112 MBytes   938 Mbits/sec    1    363 KBytes       
2018-06-20 03:46:50,775 process          L0385 DEBUG| [stdout] [ 11]   4.00-5.00   sec   113 MBytes   945 Mbits/sec    0    363 KBytes       
2018-06-20 03:46:50,775 process          L0385 DEBUG| [stdout] [ 11]   5.00-6.00   sec   112 MBytes   938 Mbits/sec    0    363 KBytes       
2018-06-20 03:46:50,775 process          L0385 DEBUG| [stdout] [ 11]   6.00-7.00   sec   113 MBytes   945 Mbits/sec    0    372 KBytes       
2018-06-20 03:46:50,776 process          L0385 DEBUG| [stdout] [ 11]   7.00-8.00   sec   112 MBytes   938 Mbits/sec    0    372 KBytes       
2018-06-20 03:46:50,776 process          L0385 DEBUG| [stdout] [ 11]   8.00-9.00   sec   113 MBytes   945 Mbits/sec    0    372 KBytes       
2018-06-20 03:46:50,776 process          L0385 DEBUG| [stdout] [ 11]   9.00-10.00  sec   112 MBytes   938 Mbits/sec    0    372 KBytes       
2018-06-20 03:46:50,776 process          L0385 DEBUG| [stdout] - - - - - - - - - - - - - - - - - - - - - - - - -
2018-06-20 03:46:50,776 process          L0385 DEBUG| [stdout] [ ID] Interval           Transfer     Bitrate         Retr
2018-06-20 03:46:50,777 process          L0385 DEBUG| [stdout] [ 11]   0.00-10.00  sec  1.10 GBytes   941 Mbits/sec    1             sender
2018-06-20 03:46:50,777 process          L0385 DEBUG| [stdout] [ 11]   0.00-10.04  sec  1.09 GBytes   936 Mbits/sec                  receiver
2018-06-20 03:46:50,777 process          L0385 DEBUG| [stdout] 
2018-06-20 03:46:50,777 process          L0385 DEBUG| [stdout] iperf Done.
2018-06-20 03:46:50,779 stacktrace       L0041 ERROR| 
2018-06-20 03:46:50,779 stacktrace       L0044 ERROR| Reproduced traceback from: /usr/local/lib/python2.7/dist-packages/avocado_framework-61.0-py2.7.egg/avocado/core/test.py:864
2018-06-20 03:46:50,779 stacktrace       L0047 ERROR| Traceback (most recent call last):
2018-06-20 03:46:50,780 stacktrace       L0047 ERROR|   File "/root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/iperf_test.py", line 103, in test
2018-06-20 03:46:50,780 stacktrace       L0047 ERROR|     % (tput, self.expected_tp))
2018-06-20 03:46:50,780 stacktrace       L0047 ERROR|   File "/usr/local/lib/python2.7/dist-packages/avocado_framework-61.0-py2.7.egg/avocado/core/test.py", line 1052, in fail
2018-06-20 03:46:50,780 stacktrace       L0047 ERROR|     raise exceptions.TestFail(message)
2018-06-20 03:46:50,780 stacktrace       L0047 ERROR| TestFail: FAIL: Throughput Actual - 941, Expected - 950
2018-06-20 03:46:50,780 stacktrace       L0048 ERROR| 
2018-06-20 03:46:50,780 test             L0869 DEBUG| Local variables:
2018-06-20 03:46:50,832 test             L0872 DEBUG|  -> cmd <type 'str'>: iperf3 -c 9.47.78.174
2018-06-20 03:46:50,832 test             L0872 DEBUG|  -> line <type 'str'>: [ 11]   0.00-10.00  sec  1.10 GBytes   941 Mbits/sec    1             sender
2018-06-20 03:46:50,832 test             L0872 DEBUG|  -> tput <type 'int'>: 941
2018-06-20 03:46:50,832 test             L0872 DEBUG|  -> result <class 'avocado.utils.process.CmdResult'>: <avocado.utils.process.CmdResult object at 0x7e978af31e10>
2018-06-20 03:46:50,832 test             L0872 DEBUG|  -> self <class 'iperf_test.Iperf'>: 1-iperf_test.py:Iperf.test;run-df57
2018-06-20 03:46:50,833 process          L0554 INFO | Running 'ssh root@9.47.78.174 "pkill iperf; rm -rf /tmp/iperf-master"'
2018-06-20 03:46:52,141 process          L0642 INFO | Command 'ssh root@9.47.78.174 "pkill iperf; rm -rf /tmp/iperf-master"' finished with 0 after 1.30632090569s
2018-06-20 03:46:52,142 test             L0289 DEBUG| DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
2018-06-20 03:46:52,142 test             L0289 DEBUG| DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
2018-06-20 03:46:52,143 test             L0289 DEBUG| DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
2018-06-20 03:46:52,143 test             L1034 ERROR| FAIL 1-iperf_test.py:Iperf.test;run-df57 -> TestFail: FAIL: Throughput Actual - 941, Expected - 950
2018-06-20 03:46:52,143 test             L1023 INFO | 
2018-06-20 03:46:58,470 job              L0498 INFO | Test results available in /root/avocado-fvt-wrapper/results/job-2018-06-20T03.46-1433909
```

**Comment by narasimhan-v** [comment]
Code inspired from https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/io/net/uperf_test.py

File: io/net/iperf_test.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    """
+    Iperf Test
+    """
+
+    def setUp(self):
+        """
+        To check and install dependencies for the test
+        """
+        smm = SoftwareManager()
+        pkgs = ["gcc", "autoconf", "perl", "m4", "git-core"]
```
Comment (PraveenPenguin) [inline io/net/iperf_test.py]: is `git-core` required

File: io/net/iperf_test.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        for pkg in pkgs:
```
Comment (PraveenPenguin) [inline io/net/iperf_test.py]: can this be `for pkg in ["gcc", "autoconf", "perl", "m4", "git-core"]:`

File: io/net/iperf_test.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                                        self.peer_ip)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.cancel("unable to copy the iperf into peer machine")
+        cmd = "ssh %s@%s \"cd /tmp/iperf-master;./configure; " \
+              "make\"" % (self.peer_user, self.peer_ip)
+        if process.system(cmd, ignore_status=True, shell=True, sudo=True):
+            self.cancel("Unable to compile Iperf into peer machine")
+        self.iperf_run = str(self.params.get("IPERF_SERVER_RUN", default=0))
+        if self.iperf_run == '1':
+            print "hello"
```
Comment (PraveenPenguin) [inline io/net/iperf_test.py]: may be left over

**Comment by narasimhan-v** [comment]
@PraveenPenguin Thanks. 
Have made the changes and pushed a new commit. Please have a check now.

**Pull request merged**